orcid-scheduler-web/src/main/java/org/orcid/scheduler/loader/cli/EmailDomainToRorLoader.java

```diff
@@ -0,0 +1,187 @@
+package org.orcid.scheduler.loader.cli;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.orcid.core.common.manager.EmailDomainManager;
+import org.orcid.core.common.manager.impl.EmailDomainManagerImpl.STATUS;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvParser;
+
+public class EmailDomainToRorLoader {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EmailDomainToRorLoader.class);
+    
+    private String filePath;
+    private EmailDomainManager emailDomainManager;
+    List<List<String>> csvData;
+
+    Set<String> invalidDomains = new HashSet<String>();  
+    
+    Map<String, DomainToRorMap> map = new HashMap<String, DomainToRorMap>();
+    
+    private int updatedEntries = 0;
+    private int createdEntries = 0;
+    
+    public EmailDomainToRorLoader(String filePath) {
+        this.filePath = filePath;
+        init(filePath);        
+    }
+    
+    public void execute() throws IOException {
+        load(this.filePath);
+        processCsvData();
+        storeDomainToRorMap();
+    }
+    
+    private void init(String filePath) {
+        Path path = Paths.get(filePath);
+        if(!Files.exists(path)) {
+            LOG.error("File does not exists: '{}'", filePath);
+            System.exit(1);
+        }
+        
+        @SuppressWarnings("resource")
+        ApplicationContext context = new ClassPathXmlApplicationContext("orcid-core-context.xml");
+        emailDomainManager = (EmailDomainManager) context.getBean("emailDomainManager");
+    }
+    
+    private void load(String filePath) throws IOException {
+        LOG.info("Reading file {}", filePath);
+        FileReader fileReader = new FileReader(filePath);       
```

Comment:
Should I the file reader be closed? https://docs.oracle.com/javase/8/docs/api/java/io/InputStreamReader.html#close--